website/server/controllers/api-v3/groups.js

```diff
@@ -561,6 +561,12 @@ api.joinGroup = {
       }
     }
 
+    if (group.type === 'guild') {
+      promises.push(User.update({
+        'achievements.guild': {$ne: true},
```

Comment:
This also will update all users. You will want to limit this to users from this guild only. Actually, I think what you want to do here is update the joining user only. 

So, if (!user. achievements.guild) user. achievements.guild = true or something